pkg/storage/stores/tsdb/cached_postings_index.go

```diff
@@ -0,0 +1,195 @@
+package tsdb
+
+import (
+	"context"
+	"encoding/binary"
+	"sort"
+	"strings"
+
+	"github.com/dennwc/varint"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	promEncoding "github.com/prometheus/prometheus/tsdb/encoding"
+
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+	"github.com/grafana/loki/pkg/storage/stores/tsdb/index"
+	"github.com/grafana/loki/pkg/util/encoding"
+)
+
+type PostingsClient interface {
```

Comment:
Suggestion: Maybe name it `PostingsReader` or `PostingsIterable`? Wdyt?

Reply:
I like `PostingsReader`, let me change it.